src/Widgets/NotificationsList.vala

```diff
@@ -77,6 +77,19 @@ public class Notifications.NotificationsList : Gtk.ListBox {
         }
     }
 
+    public void remove_entry (NotificationEntry entry) {
+        AppEntry? app_entry = null;
+
+        unowned string entry_desktop_id = entry.notification.desktop_id;
+        foreach (unowned AppEntry _app_entry in app_entries) {
+            if (_app_entry.app_info.get_id () == entry_desktop_id) {
+                app_entry = _app_entry;
+                continue;
+            }
+        }
+        app_entry.remove_notification_entry.begin (entry);
```

Comment:
Ideally, we would like to check if the app entry before use is not null here, even though it should not be.

Reply:
I don't really see a reason why app entry should be null.